lib/default.nix

```diff
@@ -64,37 +64,38 @@ let
     # linux kernel configuration
     kernel = callLibs ./kernel.nix;
 
-    inherit (builtins) add addErrorContext attrNames concatLists
-      deepSeq elem elemAt filter genericClosure genList getAttr
-      hasAttr head isAttrs isBool isInt isList isPath isString length
-      lessThan listToAttrs pathExists readFile replaceStrings seq
-      stringLength sub substring tail trace;
+    # TODO: For consistency, all builtins should also be available from a sub-library;
+    # these are the only ones that are currently not
+    inherit (builtins) addErrorContext isPath trace;
```

Comment:
Thought: 
`addErrorContext`, `trace`, `seq`, `warnIf` share a similarity that they return identity functions in terms of the returned value, adding side effects. Perhaps these should be in a sub-library together?
- `lib.nonfunctional` - they're not just functions and they tend to be about non-functional requirements
- `lib.identities` - meh, only half of the story
- `lib.sideEffects` or `lib.effectful` - maybe? technically `derivation` or `toFile` might also qualify though, if we don't equate the creation of content addressed immutable things with the construction of Nix language values, which in practice I think we all do.


Couldn't think of a great name. Maybe it's just a fairly diverse set of functions anyway.